runtime/near-vm-runner/src/runner.rs

```diff
@@ -108,3 +108,59 @@ impl VMKind {
         }
     }
 }
+
+/// Type returned by any VM instance after loading a contract and executing a
+/// function inside.
+#[derive(Debug, PartialEq)]
+pub enum VMResult {
+    /// Execution was not able to start because preconditions were not met.
+    NotRun(VMError),
+    /// Execution started but hit an error.
+    Aborted(VMOutcome, VMError),
+    /// Execution finished without error.
+    Ok(VMOutcome),
+}
+
+impl VMResult {
+    /// Consumes the `VMLogic` object and computes the final outcome with the
+    /// given error that stopped execution from finishing successfully.
+    pub fn abort(logic: VMLogic, error: VMError) -> VMResult {
+        let outcome = logic.compute_outcome_and_distribute_gas();
+        VMResult::Aborted(outcome, error)
+    }
+
+    /// Consumes the `VMLogic` object and computes the final outcome for a
+    /// successful execution.
+    pub fn ok(logic: VMLogic) -> VMResult {
+        let outcome = logic.compute_outcome_and_distribute_gas();
+        VMResult::Ok(outcome)
+    }
+
+    /// Borrow the internal outcome, if there is one.
+    pub fn outcome(&self) -> Option<&VMOutcome> {
+        match self {
+            VMResult::NotRun(_err) => None,
+            VMResult::Aborted(outcome, _err) => Some(outcome),
+            VMResult::Ok(outcome) => Some(outcome),
```

Comment:
```suggestion
            VMResult::Aborted(outcome, _err) | VMResult::Ok(outcome) => Some(outcome),
```

:+1: for `_err`, naming underscored things helps readability a bunch!

Reply:
I actually tried that first but ran into this error:

```
   --> runtime/near-vm-runner/src/runner.rs:145:48
    |
145 |             VMResult::Aborted(outcome, _err) | VMResult::Ok(outcome)=> Some(outcome),
    |                                        ----    ^^^^^^^^^^^^^^^^^^^^^ pattern doesn't bind `_err`
    |                                        |
    |                                        variable not in all patterns
```

Do you know a way of having both, the named `_err` and the succinct syntax?

Reply:
Ah, that indeed breaks the things, stupid Rust :)